lisk-dex-electron/src/Utils.js

```diff
@@ -40,6 +40,54 @@ const Keys = (dict) => Object.keys(dict);
 
 const Values = (dict) => Object.values(dict);
 
+const EstimationStatus = {
+  MATCH: 1,
+  PARTIAL_MATCH: 2,
+  NO_MATCH: 3,
+};
+
+Object.freeze(EstimationStatus);
+// considering price unit is same for both buyer and seller, for current application it's in terms of lsk per lsh
+const estimatedReturnsForSeller = (amount, price, bids) => {
+  let estimatedReturns = 0;
+  let status = EstimationStatus.NO_MATCH;
+  const maxAmountCanBeReturned = amount * price;
+  for (const bid of bids) {
+    const remainingAmount = maxAmountCanBeReturned - estimatedReturns;
```

Comment:
Note that new orders should be given the **best possible deal** based on the current state of the order book.

For example, if the ask price which is passed to this function is 0.5 but the highest bid in the list is 0.7, then the amount taken from the first bid should be `amount * 0.7` (not `amount * 0.5`) - This is a higher amount.

Instead of relying on `maxAmountCanBeReturned` to convert the ask currency into the bid currency (`ask amount -> bid amount`), you should do the inverse conversion (`bid amount -> ask amount`) - That means you need to convert the base currency from each bid into the quote currency then subtract it from the `amount` (which is in the ask currency). You can use `bid.amount / bid.price` to do the conversion in reverse.

Reply:
The other functions look good. We can remove both of the flat return ones and the related tests.